proving_system/stark/src/air/constraints/evaluator.rs

```diff
@@ -102,26 +92,26 @@ impl<'poly, F: IsFFTField, A: AIR + AIR<Field = F>> ConstraintEvaluator<'poly, F
             );
 
             let mut evaluations = self.air.compute_transition(&frame, rap_challenges);
+
+            #[cfg(debug_assertions)]
+            transition_evaluations.push(evaluations.clone());
+
             evaluations = Self::compute_constraint_composition_poly_evaluations(
                 &self.air,
                 &evaluations,
                 &divisors,
                 alpha_and_beta_transition_coefficients,
-                max_degree_power_of_two,
                 d,
             );
 
             let boundary_evaluation = zip(&boundary_polys, &boundary_zerofiers)
                 .enumerate()
                 .map(|(index, (boundary_poly, boundary_zerofier))| {
-                    let quotient_degree = boundary_poly.degree() - boundary_zerofier.degree();
-
                     let (boundary_alpha, boundary_beta) =
                         alpha_and_beta_boundary_coefficients[index].clone();
 
                     (boundary_poly.evaluate(d) / boundary_zerofier.evaluate(d))
-                        * (&boundary_alpha
-                            * d.pow(max_degree_power_of_two - (quotient_degree as u64))
+                        * (&boundary_alpha * d.pow(boundary_term_degree_adjustment)
```

Comment:
We should be able to extract `d.pow(boundary_term_degree_adjustment)` out of the closure to running exponentiation of the same values `boundary_polys.len()` times.